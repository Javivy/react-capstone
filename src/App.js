// import { BrowserRouter, Routes, Route } from 'react-router-dom';

const App = () => (
  <div>App</div>
);

export default App;
