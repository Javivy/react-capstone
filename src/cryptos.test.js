import { fireEvent, render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import store from './redux/ConfigureStore';
import App from './App';

describe('Test the Reservations', () => {
  test('Test the App component', () => {
    render(
      <Provider store={store}>
        <App />
      </Provider>,
    );
    const cryptosPage = screen.getByText('Cryptos');
    expect(cryptosPage).toBeInTheDocument();
  });

  test('Renders the cryptos page', () => {
    render(
      <Provider store={store}>
        <App />
      </Provider>,
    );
    fireEvent.click(screen.getByText('Cryptos'));
    setTimeout(() => {
      expect(screen.getByText('Bitcoin')).toBeInTheDocument();
    }, 3000);
  });

  test('Renders Bitcoin details', () => {
    render(
      <Provider store={store}>
        <App />
      </Provider>,
    );
    setTimeout(() => {
      fireEvent.click(screen.getByText('Bitcoin'));
      expect(screen.getByText('Bitcoin')).toBeInTheDocument();
    }, 3000);
  });
});

describe('Test render App', () => {
  test('Test the App component', () => {
    const rendered = render(
      <Provider store={store}>
        <App />
      </Provider>,
    );

    expect(rendered).toMatchSnapshot();
  });
});
