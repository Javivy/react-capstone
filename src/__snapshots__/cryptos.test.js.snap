// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test render App Test the App component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="navbar"
      >
        <div
          class="navbar-title"
        >
          <a
            class=""
            href="/"
          >
            <img
              alt="coin-logo"
              src="https://cryptologos.cc/logos/usd-coin-usdc-logo.png"
            />
          </a>
        </div>
        <div
          class="searcher"
        >
          <form>
            <input
              class="p-2 pr-5 rounded"
              id="search-input"
              placeholder="Search"
              style="border: 1px solid #ccc;"
              type="search"
              value=""
            />
            <button
              class="rounded-pill p-2 mx-2 bg-white "
              style="border: 1px solid #ccc;"
              type="submit"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-magnifying-glass "
                data-icon="magnifying-glass"
                data-prefix="fas"
                focusable="false"
                role="img"
                style="color: rgb(102, 102, 102);"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M416 208c0 45.9-14.9 88.3-40 122.7L502.6 457.4c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L330.7 376c-34.4 25.2-76.8 40-122.7 40C93.1 416 0 322.9 0 208S93.1 0 208 0S416 93.1 416 208zM208 352c79.5 0 144-64.5 144-144s-64.5-144-144-144S64 128.5 64 208s64.5 144 144 144z"
                  fill="currentColor"
                />
              </svg>
            </button>
          </form>
        </div>
        <div
          class="nav-links"
        >
          <a
            class="link-item"
            href="/"
          >
            Home
          </a>
          <button
            class="link-item"
            style="background: rgb(225, 245, 250);"
            type="button"
          >
            Reset
          </button>
        </div>
      </div>
      <div
        class="cryptos-container"
      />
    </div>
  </body>,
  "container": <div>
    <div
      class="navbar"
    >
      <div
        class="navbar-title"
      >
        <a
          class=""
          href="/"
        >
          <img
            alt="coin-logo"
            src="https://cryptologos.cc/logos/usd-coin-usdc-logo.png"
          />
        </a>
      </div>
      <div
        class="searcher"
      >
        <form>
          <input
            class="p-2 pr-5 rounded"
            id="search-input"
            placeholder="Search"
            style="border: 1px solid #ccc;"
            type="search"
            value=""
          />
          <button
            class="rounded-pill p-2 mx-2 bg-white "
            style="border: 1px solid #ccc;"
            type="submit"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-magnifying-glass "
              data-icon="magnifying-glass"
              data-prefix="fas"
              focusable="false"
              role="img"
              style="color: rgb(102, 102, 102);"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M416 208c0 45.9-14.9 88.3-40 122.7L502.6 457.4c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L330.7 376c-34.4 25.2-76.8 40-122.7 40C93.1 416 0 322.9 0 208S93.1 0 208 0S416 93.1 416 208zM208 352c79.5 0 144-64.5 144-144s-64.5-144-144-144S64 128.5 64 208s64.5 144 144 144z"
                fill="currentColor"
              />
            </svg>
          </button>
        </form>
      </div>
      <div
        class="nav-links"
      >
        <a
          class="link-item"
          href="/"
        >
          Home
        </a>
        <button
          class="link-item"
          style="background: rgb(225, 245, 250);"
          type="button"
        >
          Reset
        </button>
      </div>
    </div>
    <div
      class="cryptos-container"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
